src/components/AddressHistory.js

```diff
@@ -0,0 +1,227 @@
+        this.setState({ transactions, hasAfter, firstHash, lastHash });
+      }
+    }
+  }
+
+  handleDataFetched = (data, queryParams) => {
+    // Handle differently if is the first GET response we receive
+    // page indicates if was clicked 'previous' or 'next'
+    // Set first and last hash of the transactions
+    let firstHash = null;
+    let lastHash = null;
+    if (data.transactions.length) {
+      firstHash = data.transactions[0].tx_id;
+      lastHash = data.transactions[data.transactions.length-1].tx_id;
+    }
+
+    let hasAfter;
+    let hasBefore;
+    if (queryParams.page === 'previous') {
+      hasAfter = true;
+      hasBefore = data.has_more;
+      if (!hasBefore) {
+        // Went back to most recent page: clear URL params
+        this.pagination.clearQueryParams();
+      }
+    } else if (queryParams.page === 'next') {
+      hasBefore = true;
+      hasAfter = data.has_more;
+    } else {
+      // First load without parameters
+      hasBefore = false;
+      hasAfter = data.has_more;
+    }
+
+    this.setState({
+      transactions: data.transactions,
+      loading: false,
+      firstHash,
+      lastHash,
+      hasAfter,
+      hasBefore,
+      queryParams,
+    });
+  }
+
+  /**
+   * Update transactions data state after requesting data from the server
+   */
+  getData = (queryParams) => {
+    hathorLib.walletApi.getSearchAddress(this.props.address, TX_COUNT, queryParams.hash, queryParams.page, (response) => {
+      if (response.success) {
+        this.handleDataFetched(response, queryParams);
+      }
+    });
+  }
+
+
+  render() {
+    if (this.state.transactions.length === 0) {
+      return <p>This address does not have any transactions yet.</p>
+    }
+
+    const loadPagination = () => {
+      if (this.state.transactions.length === 0) {
+        return null;
+      } else {
+        return (
+          <nav aria-label="Tx pagination" className="d-flex justify-content-center">
+            <ul className="pagination">
+              <li ref="txPrevious" className={(!this.state.hasBefore || this.state.transactions.length === 0) ? "page-item mr-3 disabled" : "page-item mr-3"}>
+                <Link className="page-link" to={this.pagination.paginationUrl({hash: this.state.firstHash, page: 'previous'})}>Previous</Link>
+              </li>
+              <li ref="txNext" className={(!this.state.hasAfter || this.state.transactions.length === 0) ? "page-item disabled" : "page-item"}>
+                <Link className="page-link" to={this.pagination.paginationUrl({hash: this.state.lastHash, page: 'next'})}>Next</Link>
+              </li>
+            </ul>
+          </nav>
+        );
+      }
+    }
+
+    const loadTable = () => {
+      return (
+        <div className="table-responsive mt-5">
+          <table className="table table-striped" id="tx-table">
+            <thead>
+              <tr>
+                <th className="d-none d-lg-table-cell">Type</th>
```

Comment:
Missing i18n markers.

Reply:
We don't have i18n in the explorer, should we do something?

Reply:
No. Sorry.